js/interface.js

```diff
@@ -39,6 +40,18 @@ $editInput
 $('body')
   .keydown(editImageTitleBinding);
 
+function restoreFromMetadata() {
+  var metadataExists = data.images.some(function(elem) {
+    return elem.masterMediaFileId;
```

Comment:
You don't seem to check for which metadata is for which image? I would have expected some code like this:

```
foreach data.images
  - find relevant metadata
  - if found, assign it to image
```

Reply:
@squallstar 
My idea is that we check if metadata exists and then assign it by index as both old images and duplicated images have the same index. Do you think we need another solution?

Reply:
Is there any reason on why you don't want to use a better approach like finding the correct record by matching for an ID?

Reply:
I feel that this way would be safer and easier. At first I tried doing it by matching IDs but apparently after you duplicate the app, images receive masterID which also have to consider. I remember that it resulted in many if checks and loops. We decided to try this approach which seem to work fine.

Reply:
1. How is index based match safer than an actual ID check? 
1b. You should be able to match for either `id` or `masterMediaFileId`
2. Easier doesn't necessarily mean better. Same applies for complexity: additional complexity (e.g. loops) may be required to ensure the correct title is applied to the correct image.

Reply:
@squallstar 
I will remake then to match images by ID.

Reply:
@squallstar 
While I was changing to make it work by ID I remembered what was the issue with it.
Match by ID will not work if we make copy of an app and then make the copy of copy of an app because of next. 
For example, 
1. App has 2 images with IDs 1 and 2. We save them to metadata.
2. We duplicate App. Copy of an app now has IDs 3 and 4. MasterMediaFileIds are 1 and 2. So far everything works. We restore titles by matching metadata and masterMediaFileIds. 
3. We duplicate a copy of App straight away without opening Copy App. 
IDs would be 5 and 6 and masterAppIds would be 3 and 4. Now we cannot restore titles because metadata and masterAppIds are different.

Reply:
I see, OK that makes sense. In that case I understand, let's then match by index as you initially suggested.

Reply:
Is the PR ready for review or do you need further changes?

Reply:
@squallstar 
I made some small fixes. This PR is ready for review.